File: modules/DumbPosTagger.lua
require 'torch'
require 'math'
require 'nn'

dofile 'PosTagger.lua'

local DumbPosTagger, parent = torch.class('nn.DumbPosTagger', 'nn.PosTagger')

-- A dumb PosTagger that just returns {'NN', 'NN', ...} and -infinity.
function DumbPosTagger:__init()
end

function DumbPosTagger:train(tagged_sentences)
  -- do nothing
end

function DumbPosTagger:validate(tagged_sentences)
  -- do nothing
end

function DumbPosTagger:tag(sentence)
  local tags = {}
  for i = 1, #sentence do
    tags[i] = 'NN'
  end
  return tags
end

function DumbPosTagger:scoreTagging(tagged_sentence)
  return -math.huge
end

File: theano_port/DumbPosTagger.py
from theano_port.PosTagger import PosTagger

class DumbPosTagger(PosTagger):
  
  def train(self, tagged_sentences, learning_rate, iterations):
    pass

  def validate(self, tagged_sentences):
    pass

  def tag(self, sentence):
    return ['NN' for i in xrange(0, len(sentence))]

  def score_tagging(self, tagged_sentence):
    return float('-inf')
